sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -71,6 +71,57 @@ def _succeeded(status):
     return str(status).lower() in _SUCCEEDED
 
 
+class OperationState(ABC):
+    @abc.abstractproperty
+    def finished(self):
+        # type: () -> list(str)
+        raise NotImplementedError()
+
+    @abc.abstractproperty
+    def failed(self):
+        # type: () -> list(str)
+        raise NotImplementedError()
+
+    @abc.abstractproperty
+    def succeeded(self):
+        # type: () -> list(str)
+        raise NotImplementedError()
+
+    def has_finished(self, status):
```

Comment:
II think in the `ABC`, all you need to have are the properties for `finished` etc. I don't think you should expose the ability to override the `has_finished` check, since (at least as far as i foresee), that check really doesn't need to be changed ever.

This way, you can change the module level functions at the top to take in both a status, and an operation state property, so it would look like:

```
def _finished(operation_state, status):
    if hasattr(status, "value"):
        status = status.value
    return str(status).lower() in operation_state.finished
```

You can also delete the variable `_FINISHED` etc from the code at the tope of the module

Reply:
Thanks for the review!  I know it's been a while but I finally got around to addressing these comments.  :)  

I've made the changes you mentioned here.  Let me know if I missed anything.